Sources/Utility/Toolchain.swift

```diff
@@ -0,0 +1,58 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+ This file defines support functions for resources tied to the installed
+ location of the containing product binary.
+ */
+
+import func POSIX.getenv
+import func POSIX.popen
+
+public protocol Installation {
+    static func which(cmd: String) -> String
+}
+
+public struct Toolchain: Installation {
+    public static func which(cmd: String) -> String {
+        if let exepath = Process.arguments.first {
+            if let path = try? Path.join(exepath, "..", cmd).abspath() where path.isFile {
+                return path
+            }
+        }
+
+    #if os(OSX)
+        if let cmdpath = (try? popen(["xcrun", "--find", cmd]))?.chuzzle() {
+            return cmdpath
+        }
+    #endif
+        
+        return cmd
+    }
+
+    //TODO better
+    public static var prefix: String {
+        return swiftc.parentDirectory.parentDirectory.parentDirectory
+    }
+
+    /// the location of swiftc relative to our installation
+    public static let swiftc = getenv("SWIFT_EXEC") ?? Toolchain.which("swiftc")
```

Comment:
This gives me wrong path for `swiftc`

```bash
➜ xcrun --find swiftc
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc
➜ xcrun --find swift-build-tool
/Library/Developer/Toolchains/swift-latest.xctoolchain/usr/bin/swift-build-tool
```

Causing the error 
```
/Users/bhargavg/Documents/workspaces/xcode/github/swift-package-manager/Package.swift:11:8: error: module file was created by a newer version of the compiler: /Users/bhargavg/Documents/workspaces/xcode/github/swift-package-manager/.build/lib/swift/pm/PackageDescription.swiftmodule
```
when I try to run `.build/debug/swift-build --generate-xcodeproj`